### PR TITLE
feat: --provider flag #139

### LINKED ISSUE #139: --provider flag
`jspm install --provider skypack` etc would be really nice to support. It's one argument to carry through I think we should push for this for the release.

### PR DESCRIPTION
**Pull request by Aslemammad**
Resolves #139

### CODE REVIEW

**Review by guybedford — APPROVED** [review]
Thanks. Can we just add a test.

File: test/install.test.ts [hunk 1]
```diff
@@ -124,3 +124,18 @@ import install from '../src/install'
 
   await fs.rm('test/importmap.modified.json')
 }
+
+{
+  /* provider */
```
Comment (Aslemammad) [inline test/install.test.ts:129]: Here's a test.

File: test/install.test.ts [hunk 2]
```diff
@@ -124,3 +124,18 @@ import install from '../src/install'
...
+  await install(['react@17.0.1', 'react-dom@17.0.1'], {
+    env: 'production,browser',
+    provider: 'jspm',
```
Comment (guybedford) [inline test/install.test.ts:132]: `provider: 'jsdelivr'`

File: test/install.test.ts [hunk 3]
```diff
@@ -124,3 +124,18 @@ import install from '../src/install'
...
+    map: 'test/importmap.json',
+  })
+  const map = JSON.parse(
+    await fs.readFile('test/importmap.json', 'utf-8'),
+  )
+  assert.deepEqual(
+    map.provider, 'jspm',
```
Comment (guybedford) [inline test/install.test.ts:139]: This is wrong, the provider is automatically detected by JSPM generator, and if not that is a bug.

**Comment by Aslemammad** [comment]
It's good to merge, but failing due to the generator traceInstall error (fixed, but not released yet as I see)

**Pull request merged**